### PR TITLE
Release v23

### PR DESCRIPTION
**Pull request by JuiP**
- [x] Fix read_file method for resume from journal

- [x] Add screenshots of solutions all levels, could be used for adding a help button functionality later.

- [x] Other changes to release the activity

### CODE REVIEW

**Comment by JuiP** [comment]
@srevinsaju @Saumya-Mishra9129 @chimosky @quozl Please test and review changes so I can go ahead with the release :)

File: IQ.py [hunk 1]
```diff
@@ -349,17 +349,17 @@ def shuffle(self, lst):
             l1.remove(lst[r])
         return lt
 
-    def centre_blit(self, screen, img, xxx_todo_changeme, angle=0):  # rotation is clockwise
-        (cx, cy) = xxx_todo_changeme
+    def centre_blit(self, screen, img, coordinates, angle=0):  # rotation is clockwise
+        (cx, cy) = coordinates
```
Comment (srevinsaju) [inline IQ.py]: redundant parenthesis may be removed (optional)

File: IQ.py [hunk 2]
```diff
@@ -349,17 +349,17 @@ def shuffle(self, lst):
...
             img1 = pygame.transform.rotate(img, -angle)
         rect = img1.get_rect()
         screen.blit(img1, (cx-rect.width/2, cy-rect.height/2))
 
-    def text_blit(self, screen, s, font, xxx_todo_changeme1, xxx_todo_changeme2, shadow=True):
-        (cx, cy) = xxx_todo_changeme1
-        (r, g, b) = xxx_todo_changeme2
+    def text_blit(self, screen, s, font, coordinates, color, shadow=True):
+        (cx, cy) = coordinates
+        (r, g, b) = color
```
Comment (srevinsaju) [inline IQ.py]: same

File: IQ.py [hunk 3]
```diff
@@ -373,9 +373,9 @@ def text_blit(self, screen, s, font, xxx_todo_changeme1, xxx_todo_changeme2, sha
         screen.blit(text, rect)
         return rect
 
-    def text_blit1(self, screen, s, font, xxx_todo_changeme3, xxx_todo_changeme4, shadow=True):
-        (x, y) = xxx_todo_changeme3
-        (r, g, b) = xxx_todo_changeme4
+    def text_blit1(self, screen, s, font, coordinates, color, shadow=True):
+        (x, y) = coordinates
+        (r, g, b) = color
```
Comment (srevinsaju) [inline IQ.py]: same

File: IQ.py [hunk 4]
```diff
@@ -393,8 +393,8 @@ def text_blit1(self, screen, s, font, xxx_todo_changeme3, xxx_todo_changeme4, sh
     # d is the # of pixels in the border around the text
     # (cx,cy) = co-ords centre - (0,0) means use screen centre
 
-    def message(self, screen, font, m, xxx_todo_changeme5=(0, 0), d=20):
-        (cx, cy) = xxx_todo_changeme5
+    def message_func(self, screen, font, m, coordinates=(0, 0), d=20):
+        (cx, cy) = coordinates
```
Comment (srevinsaju) [inline IQ.py]: this

**Review by srevinsaju — COMMENTED** [review]
I found minor changes; Its entirely optional. I will do activity testing tomorrow. Great work!!

File: IQ.py (hunk 4 above)
Comment (JuiP) [reply IQ.py]: Done in ad31fc5

File: IQ.py (hunk 3 above)
Comment (JuiP) [reply IQ.py]: Done in ad31fc5

File: IQ.py (hunk 2 above)
Comment (JuiP) [reply IQ.py]: Done in ad31fc5

File: IQ.py (hunk 1 above)
Comment (JuiP) [reply IQ.py]: Thanks for noticing! :)
Done in ad31fc5

**Review by Saumya-Mishra9129 — COMMENTED** [review]
Reviewed and Tested, works fine. Nice work.

**Review by srevinsaju — COMMENTED** [review]
Good work; works fine; Actually some regressions which were made during my port which I found now... smh

* Mouse cursor is not drawn when cursor moved out of the Sugargame canvas
image

* Occasional error: 
```
Traceback (most recent call last):
  File "/run/media/srevinsaju/ARCH_DATA/git/iq-activity/IQ.py", line 245, in run
    self.do_key(event.key)
  File "/run/media/srevinsaju/ARCH_DATA/git/iq-activity/IQ.py", line 92, in do_key
    if key in self.SQUARE:
AttributeError: 'IQ' object has no attribute 'SQUARE'
```
It happens rarely; One way to reproduce might be to wait for sugar desktop to sleep (the screen goes black) and move the mouse to wake it again. the error is possibly to occur. I will update the comment if I find a method to reproduce it.

**Review by srevinsaju — APPROVED** [review]
Tested as of https://github.com/sugarlabs/iq-activity/pull/20/commits/0b348fa6986bc8524f35df2e8d62424180e18c2a; works perfectly!

**Comment by JuiP** [comment]
Thanks a lot for reviewing and testing! @srevinsaju @Saumya-Mishra9129 

Done! Ready for release :)
@quozl @chimosky

**Comment by quozl** [comment]
Thanks.  Reviewed.  Approved.

**Comment by chimosky** [comment]
Tested, thanks.

**Pull request merged**